Examples/Algorithms/TrackFinding/src/TrackParamsLookupEstimation.cpp

```diff
@@ -0,0 +1,101 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/TrackFinding/TrackParamsLookupEstimation.hpp"
+
+#include "Acts/EventData/SourceLink.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+
+ActsExamples::TrackParamsLookupEstimation::TrackParamsLookupEstimation(
+    const Config& config, Acts::Logging::Level level)
+    : IAlgorithm("TrackParamsLookupEstimation", level),
+      m_cfg(std::move(config)) {
+  // Iterate over the reference layers and create
+  // track parameter accumulators
+  for (const auto& [geoId, refSurface] : m_cfg.refLayers) {
+    // Get bounds to construct the accumulator grid
+    auto bounds =
+        dynamic_cast<const Acts::RectangleBounds*>(&refSurface->bounds());
+
+    if (bounds == nullptr) {
+      throw std::invalid_argument("Only rectangle bounds supported");
+    }
+
+    // Initialize the accumulator grid
+    auto halfX = bounds->halfLengthX();
+    auto halfY = bounds->halfLengthY();
+
+    LookupAxisGen axisGen{
+        {-halfX, halfX}, m_cfg.bins.first, {-halfY, halfY}, m_cfg.bins.second};
+
+    TrackParamsLookupAccumulator::Config accConfig{axisGen};
+
+    // Each reference layer has its own accumulator
+    m_accumulators[geoId] =
+        std::make_unique<TrackParamsLookupAccumulator>(accConfig);
+  }
+
+  m_inputParticles.initialize(m_cfg.inputParticles);
+  m_inputSimHits.initialize(m_cfg.inputHits);
+}
+
+ActsExamples::TrackParamsLookupEstimation::~TrackParamsLookupEstimation() {
```

Comment:
Can you do this in the `finalize` method instead of the destructor?

Reply:
Done